### PR TITLE
DepartureTransportMeans feature branch

### PR DESCRIPTION
**Pull request by justega247**

### CODE REVIEW

**Comment by justega247** [comment]
rebuild

File: app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala [hunk 1]
```diff
@@ -77,11 +90,20 @@ case class TransitionTransportMeansDepartureDomain(
...
 
 object TransitionTransportMeansDepartureDomain {
 
+  def asString(identification: Option[Identification], identificationNumber: Option[String], index: Index)(implicit messages: Messages): String =
+    (identification, identificationNumber) match {
+      case (Some(id), Some(idNumber)) => messages("departureTransportMeans.label.bothArgs", index.display, id.asString, idNumber)
+      case (Some(id), None)           => messages("departureTransportMeans.label.oneArg", index.display, id.asString)
+      case (None, Some(idNumber))     => messages("departureTransportMeans.label.oneArg", index.display, idNumber)
+      case _                          => messages("departureTransportMeans.label.noArgs", index.display)
+    }
```
Comment (jackheslop96) [inline app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala:105]: Bit of duplicated code here but no biggy. Could look to incorporate this with app/models/TransportMeans.scala but can do so in a separate PR

File: app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala [hunk 2]
```diff
@@ -124,3 +146,27 @@ object TransitionTransportMeansDepartureDomain {
...
+    extends TransportMeansDepartureDomain {
+
+  override def asString(implicit messages: Messages): String =
+    TransitionTransportMeansDepartureRulesDisabledDomain.asString(identification, identificationNumber, index)
+}
+
+object TransitionTransportMeansDepartureRulesDisabledDomain {
+
+  def asString(identification: Identification, identificationNumber: String, index: Index)(implicit messages: Messages): String =
+    messages("departureTransportMeans.label.bothArgs", index.display, identification.asString, identificationNumber)
```
Comment (jackheslop96) [inline app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala:164]: As above

**Review by jackheslop96 — APPROVED** [review]
